ladybug/src/main/java/org/frankframework/ibistesttool/jdbc/LadyBugDataSourceFactory.java

```diff
@@ -19,28 +19,18 @@
 import javax.sql.DataSource;
 import javax.sql.XADataSource;
 
-import org.apache.commons.lang3.StringUtils;
-import org.frankframework.jdbc.datasource.DataSourceFactory;
-import org.frankframework.jta.narayana.NarayanaDataSource;
-import org.springframework.beans.BeansException;
-import org.springframework.context.ApplicationContext;
-import org.springframework.context.ApplicationContextAware;
+import org.frankframework.jdbc.datasource.PoolingDataSourceFactory;
 
-public class LadyBugDataSourceFactory extends DataSourceFactory implements ApplicationContextAware {
-	private String txManagerType;
-
-	@Override
-	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
-		String txManager = applicationContext.getEnvironment().getProperty("application.server.type.custom");
-		if(StringUtils.isNotEmpty(txManager)) {
-			txManagerType = txManager.toUpperCase();
-		}
-	}
+/**
+ * The Ladybug uses it's own factory so it can be in control over it's connections.
+ * The DataSource is wrapped in a TransactionAwareDataSourceProxy.
+ */
+public class LadyBugDataSourceFactory extends PoolingDataSourceFactory {
 
 	@Override
 	protected DataSource augmentDatasource(CommonDataSource dataSource, String dataSourceName) {
-		if("NARAYANA".equals(txManagerType) && dataSource instanceof XADataSource source) {
-			return new NarayanaDataSource(source, dataSourceName);
+		if(dataSource instanceof XADataSource) {
```

Comment:
This will break.

I've been there, done that, got to keep both parts.

There are plenty of databases which have a datasource that implements both the interfaces `DataSource` _and_ `XADataSource`.

Reply:
I'm well aware, but the ladybug does not work with XA Resources. Wrapping them in Narayana is a bad idea as it will mix with out transactions...

Reply:
Yes you can, with the appropriate (non-xa) driver 😄